### ISSUE TITLE
expected scalar type BFloat16 but found Float

### ISSUE THREAD

**Issue by jhautefeuille** [brycedrennan/imaginAIry#81]
```bash
julien@pop7550:~$ imagine "a scenic landscape" "a photo of a dog" "photo of a fruit bowl" "portrait photo of a freckled woman"
🤖🧠 imaginAIry received 4 prompt(s) and will repeat them 1 times to create 4 images.
Running in CPU mode. it's gonna be slooooooow.
Generating 🖼  1/4: "a scenic landscape" 512x512px seed:913183355 prompt-strength:7.5 steps:40 sampler-type:plms
Loading model /home/julien/.cache/huggingface/transformers/d12e71b67e29abaf317bf9d0e31644872fd2072509a4b3582cbb0c30f70824e9.98fc1312797017a8bac6993df565908fd18f09319b40d9bd35457dfa1459ecf0 onto cpu backend...
  0%|                                                                            | 0/40 [00:04<?, ?it/s]
/usr/lib/python3/dist-packages/apport/report.py:13: DeprecationWarning: the imp module is deprecated in favour of importlib and slated for removal in Python 3.12; see the module's documentation for alternative uses
  import fnmatch, glob, traceback, errno, sys, atexit, locale, imp, stat
Traceback (most recent call last):
  File "/home/julien/.local/bin/imagine", line 8, in <module>
    sys.exit(imagine_cmd())
  File "/usr/lib/python3/dist-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/julien/.local/lib/python3.10/site-packages/imaginairy/cmds.py", line 238, in imagine_cmd
    imagine_image_files(
  File "/home/julien/.local/lib/python3.10/site-packages/imaginairy/api.py", line 74, in imagine_image_files
    for result in imagine(
  File "/home/julien/.local/lib/python3.10/site-packages/imaginairy/api.py", line 248, in imagine
    samples = sampler.sample(
  File "/home/julien/.local/lib/python3.10/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/home/julien/.local/lib/python3.10/site-packages/imaginairy/samplers/plms.py", line 98, in sample
    noisy_latent, predicted_latent, noise_pred = self.p_sample_plms(
  File "/home/julien/.local/lib/python3.10/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/home/julien/.local/lib/python3.10/site-packages/imaginairy/samplers/plms.py", line 139, in p_sample_plms
    noise_pred = get_noise_prediction(
  File "/home/julien/.local/lib/python3.10/site-packages/imaginairy/samplers/base.py", line 130, in get_noise_prediction
    noise_pred_neutral, noise_pred_positive = denoise_func(
  File "/home/julien/.local/lib/python3.10/site-packages/imaginairy/modules/diffusion/ddpm.py", line 763, in apply_model
    x_recon = self.model(x_noisy, t, **cond)
  File "/home/julien/.local/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1190, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/julien/.local/lib/python3.10/site-packages/imaginairy/modules/diffusion/ddpm.py", line 888, in forward
    out = self.diffusion_model(x, t, context=cc)
  File "/home/julien/.local/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1190, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/julien/.local/lib/python3.10/site-packages/imaginairy/modules/diffusion/openaimodel.py", line 778, in forward
    h = module(h, emb, context)
  File "/home/julien/.local/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1190, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/julien/.local/lib/python3.10/site-packages/imaginairy/modules/diffusion/openaimodel.py", line 85, in forward
    x = layer(x, context)
  File "/home/julien/.local/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1190, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/julien/.local/lib/python3.10/site-packages/imaginairy/modules/attention.py", line 320, in forward
    x = block(x, context=context)
  File "/home/julien/.local/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1190, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/julien/.local/lib/python3.10/site-packages/imaginairy/modules/attention.py", line 266, in forward
    return checkpoint(
  File "/home/julien/.local/lib/python3.10/site-packages/imaginairy/modules/diffusion/util.py", line 150, in checkpoint
    return CheckpointFunction.apply(func, len(inputs), *args)
  File "/home/julien/.local/lib/python3.10/site-packages/imaginairy/modules/diffusion/util.py", line 163, in forward
    output_tensors = ctx.run_function(*ctx.input_tensors)
  File "/home/julien/.local/lib/python3.10/site-packages/imaginairy/modules/attention.py", line 272, in _forward
    x = self.attn1(self.norm1(x)) + x
  File "/home/julien/.local/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1190, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/julien/.local/lib/python3.10/site-packages/torch/nn/modules/normalization.py", line 190, in forward
    return F.layer_norm(
  File "/home/julien/.local/lib/python3.10/site-packages/imaginairy/utils.py", line 102, in _fixed_layer_norm
    return torch.layer_norm(
RuntimeError: expected scalar type BFloat16 but found Float
```

**Comment by pirboazo** [comment]
Hi,
running on cpu mode.
> Generating 🖼  1/1: "a close-up portrait of a cat" 512x512px seed:173876548 prompt-strength:7.5 steps:40 sampler-type:plms

same stack of error.

work fine with stableDiffusion

**Comment by brycedrennan** [comment]
What version of torch are you both on?

**Comment by jbenech** [comment]
Hello,

I have the same problem on a Windows configuration. My torch version is: 1.13.0

As attachment, the pipdeptree for imagineairy: [pipdeptree-imagineairy.txt](https://github.com/brycedrennan/imaginAIry/files/9968468/pipdeptree-imagineairy.txt)

**Comment by pirboazo** [comment]
Hi 
I use pipenv , it's 1.13.0

[imaginairy.txt](https://github.com/brycedrennan/imaginAIry/files/9969086/imaginairy.txt)

**Comment by brycedrennan** [comment]
Confirmed.

**Comment by brycedrennan** [comment]
Fixed in 5.0.1 (just released)
